### PR TITLE
Feature/call dwp api

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

**Comment by dwybourn** [comment]
This PR is not a complete solution as we still need to fetch properties from configuration to pass onto the dwp api. The PR has been created as the work done so far was getting quite large.

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityController.java [hunk 1]
```diff
@@ -1,37 +1,39 @@
...
     /**
-     * Invokes downstream services to obtain a decision on eligibility for the given request,
+     * Invokes downstream services to obtain a decision on eligibility for the given person,
      * persists the request and decision with the claimant service.
      *
-     * @param request the {@link EligibilityRequest}
+     * @param person the {@link PersonDTO}
      * @return the {@link EligibilityResponse}
      */
     @PostMapping(path = "/eligibility", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityController.java]: Should be `v1/eligibility`

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import static org.mockito.BDDMockito.given;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.dhsc.htbhf.eligibility.helper.PersonTestFactory.aPerson;
+import static uk.gov.dhsc.htbhf.eligibility.model.EligibilityStatus.ELIGIBLE;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class EligibilityControllerTest {
+
+    private static final String ENDPOINT_URL = "/eligibility";
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java]: Should be `v1/eligibility`

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/helper/PersonTestFactory.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package uk.gov.dhsc.htbhf.eligibility.helper;
+
+import uk.gov.dhsc.htbhf.eligibility.model.AddressDTO;
+import uk.gov.dhsc.htbhf.eligibility.model.PersonDTO;
+
+import java.time.LocalDate;
+
+public class PersonTestFactory {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/eligibility/helper/PersonTestFactory.java]: I'd call this `PersonDTOTestDataFactory` as we have more than one person model object in this project.

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/service/EligibilityServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private PersonDTOToDWPPersonConverter converter;
+    @MockBean
+    private DWPClient dwpClient;
+
+    @Test
+    void shouldCreateRequest() {
+        PersonDTO person = aPerson();
+        DWPPersonDTO dwpPerson = DWPPersonDTO.builder().build();
+        given(converter.convert(person)).willReturn(dwpPerson);
+        EligibilityResponse eligibilityResponse = EligibilityResponse.builder().eligibilityStatus(ELIGIBLE).build();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/eligibility/service/EligibilityServiceTest.java]: I'd create a Test Data Factory for Eligibility Responses as you've duplicated this code a few times in this PR.

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/helper/PersonTestFactory.java (hunk 3 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/eligibility/helper/PersonTestFactory.java]: Renamed.

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/service/EligibilityServiceTest.java (hunk 4 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/eligibility/service/EligibilityServiceTest.java]: Created test factory and used throughout tests.

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java (hunk 2 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java]: Updated.

File: api/src/main/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityController.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityController.java]: Updated.

**Pull request merged**